_posts/en/newsletters/2021-07-14-newsletter.md

```diff
@@ -0,0 +1,208 @@
+---
+title: 'Bitcoin Optech Newsletter #157'
+permalink: /en/newsletters/2021/07/14/
+name: 2021-07-14-newsletter
+slug: 2021-07-14-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter summarizes a discussion about a proposed new
+opcode and links to an updated wiki page for tracking bech32m support.
+Also included are our regular sections with highlights from a Bitcoin
+Core PR Review Club meeting, suggestions about preparing for taproot,
+and descriptions of notable changes to popular Bitcoin infrastructure
+projects.
+
+## News
+
+- **Request for `OP_CHECKSIGFROMSTACK` design suggestions:** Jeremy
+  Rubin [posted][rubin csfs] to the Bitcoin-Dev mailing list a draft specification
+  for an [OP_CHECKSIGFROMSTACK][topic op_checksigfromstack] opcode and
+  asked for feedback from any developers who would prefer an alternative
+  design.  Some alternatives were discussed, but the thread also
+  branched off into a discussion about whether an [OP_CAT][] opcode
+  should be introduced at the same time.
+
+    `OP_CAT` and `OP_CSFS` would enable arbitrary transaction
+    introspection---the ability to receive bitcoins to a script that
+    could check almost any part of the transaction that later spends
+    those bitcoins.  This can enable many advanced features (including
+    versions[^expensive] of other proposed upgrades like
+    [SIGHASH_ANYPREVOUT][topic sighash_noinput] and
+    [OP_CHECKTEMPLATEVERIFY][topic op_checktemplateverify]), but
+    `OP_CAT` also makes it possible to create recursive
+    [covenants][topic covenants] which could permanently restrict the
+    spendability of any bitcoins committed to the covenant.  Some people
+    have objected to allowing covenants in Bitcoin, but several
```

Comment:
Are there any objections from the thread that help inform the debate for the reader that could be linked here?